pymeasure/instruments/siglenttechnologies/siglent_spdbase.py

```diff
@@ -0,0 +1,278 @@
+        values={False: "2W", True: "4W"},
+        map_values=True
+    )
+
+    def save_config(self, index):
+        """Save the current config to memory.
+
+        :param index:
+            string: index of the location to save the configuration
+        :returns: self
+        """
+        self.save_settings = index
+
+    def recall_config(self, index):
+        """Recall a config from memory.
+
+        :param index:
+            string: index of the location from which to recall the configuration
+        :returns: self
+        """
+        self.recall_settings = index
+
+
+class SPDChannel(object):
+    """ The channel class for Siglent SPDxxxxX instruments.
+    """
+
+    def __init__(self, instrument, channel: int = 1):
+        self.instrument = instrument
+        self.channel = channel
+
+    voltage = Instrument.measurement(
+        "MEAS:VOLT? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    current = Instrument.measurement(
+        "MEAS:CURR? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    power = Instrument.measurement(
+        "MEAS:POWE? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    set_current = Instrument.control(
+        "CH{channel}:CURR?", "CH{channel}:CURR %g",
+        """Control the output current configuration of the channel.
+
+        :type: float
+
+        """,
+        validator=truncated_range,
+        values=[0, 8],
+        dynamic=True
+    )
+
+    set_voltage = Instrument.control(
+        "CH{channel}:VOLT?", "CH{channel}:VOLT %g",
+        """Control the output voltage configuration of the channel.
+
+        :type: float
+
+        """,
+        validator=truncated_range,
+        values=[0, 16],
+        dynamic=True
+    )
+
+    output = Instrument.setting(
+        "OUTP CH{channel},%s",
+        """Configure the power supply output.
+
+        :type: bool
+            ``True``: disable local interface
+            ``False``: enables local interface
+
+        """,
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True
+    )
+
+    enable_timer = Instrument.setting(
+        "TIME CH{channel},%s",
+        """Enable the channel timer.
+
+        :type: bool
+        """,
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True
+    )
+
+    def ask(self, cmd):
+        return self.instrument.ask(cmd.format(channel=self.channel))
+
+    def write(self, cmd):
+        self.instrument.write(cmd.format(channel=self.channel))
+
+    def values(self, cmd, **kwargs):
+        """ Reads a set of values from the instrument through the adapter,
+        passing on any key-word arguments.
+        """
+        return self.instrument.values(cmd.format(channel=self.channel))
```

Comment:
You say, you pass on kwargs, but you don't do it.
```suggestion
        """ Reads a set of values from the instrument through the adapter,
        passing on any key-word arguments.
        """
        return self.instrument.values(cmd.format(channel=self.channel), **kwargs)
```

Reply:
Good spot